index.js

```diff
@@ -13,89 +13,107 @@
+ */
+function findRuntimePath() {
+    const upPkg = readPkgUp.sync();
+    // case 1: we're in NR itself
+    if (upPkg.pkg.name === 'node-red') {
+        return path.join(path.dirname(upPkg.path), upPkg.pkg.main);
+    }
+    // case 2: NR is installed alongside node-red-node-test-helper
+    if ((upPkg.pkg.dependencies && upPkg.pkg.dependencies['node-red'])
+        || (upPkg.pkg.devDependencies && upPkg.pkg.devDependencies['node-red'])) {
+        const dirpath = path.join(path.dirname(upPkg.path), 'node_modules', 'node-red');
     try {
-        runtimePath = require.resolve('node-red');
-        initRuntime(runtimePath);
-    } catch (err) {
-        // no runtime path - init must be called from test
+            const pkg = require(path.join(dirpath, 'package.json'));
+            return path.join(dirpath, pkg.main);
+        } catch (ignored) {}
     }
 }
 
-function initRuntime(requirePath) {
-
-    try {
-        RED = require(requirePath);
+class NodeTestHelper extends EventEmitter {
+    constructor() {
+        super();
 
-        // public runtime API
-        redNodes = RED.nodes;
-        events = RED.events;
-        log = RED.log;
+        this._sandbox = sinon.createSandbox();
 
-        // access internal Node-RED runtime methods
-        var prefix = requirePath.substring(0, requirePath.indexOf('/red.js'));
-        context = require(prefix+"/runtime/nodes/context");
-        comms = require(prefix+"/api/editor/comms");
-        credentials = require(prefix+"/runtime/nodes/credentials");
+        this._address = '127.0.0.1';
+        this._listenPort = 0; // ephemeral
 
-    } catch (err) {
-        // ignore, assume init will be called again by a test script supplying the runtime path
+        this.init();
     }
-}
-
-initRuntime(runtimePath);
-
-var app = express();
-
-var address = '127.0.0.1';
-var listenPort = 0; // use ephemeral port
-var port;
-var url;
-var logSpy;
-var server;
 
-function helperNode(n) {
-    RED.nodes.createNode(this, n);
-}
+    _initRuntime(requirePath) {
+        try {
+            const RED = this._RED = require(requirePath);
+
+            // public runtime API
+            this._redNodes = RED.nodes;
+            this._events = RED.events;
+            this._log = RED.log;
+
+            // access internal Node-RED runtime methods
+            const prefix = path.dirname(requirePath);
+            this._context = require(path.join(prefix, 'runtime', 'nodes', 'context'));
+            this._comms = require(path.join(prefix, 'api', 'editor', 'comms'));
+
+            this.credentials = require(path.join(prefix, 'runtime', 'nodes', 'credentials'));
+
+            // proxy the methods on Node.prototype to both be Sinon spies and asynchronously emit
+            // information about the latest call
+            const NodePrototype = require(path.join(prefix, 'runtime', 'nodes', 'Node')).prototype;
+            PROXY_METHODS.forEach(methodName => {
```

Comment:
I suggest moving this code where you proxy Node methods to the `helper.load()` method where the `RED.log` is proxied.  This is more consistent, and we can then call `this.sandbox.restore()` on `unload`, to restore the proxied methods and the runtime logSpy in one go.